### PR TITLE
Added DTOs for our Event.

### LINKED ISSUE #143: Create DTOs for Event
📝 Description:
Implement all required Data Transfer Objects (DTOs) to support the "Create Event" feature. These DTOs will be used for request validation, form population, and response mapping between the client and backend.

📌 DTOs to Create:

1.CreateEventRequestDto
Used to receive data when creating a new event (client → server)

Fields:

title: String — required, max 70 characters

description: String — required, 20 to 63,206 characters

visibility: Visibility (enum: OPEN, CLOSED) — required

eventTypes: Set<EventType> (must include at least one: PLACE and/or ONLINE)

locations: List<LocationDto> — required if type includes PLACE

onlineLinks: List<String> — required if type includes ONLINE

eventDateTimes: List<EventDateTimeDto> — required, 1–7 future dates

images: List<ImageDto> — optional, up to 5

mainImageId: Long — optional (ID of main image)

2.EventResponseDto
Returned after successfully creating an event

Fields:

eventId: Long

title: String

createdAt: LocalDateTime

3.EditEventRequestDto
Used to update an existing event (client → server)

Fields:

eventId: Long — required

title: String

description: String

visibility: Visibility (enum)

eventTypes: Set<EventType>

locations: List<LocationDto>

onlineLinks: List<String>

eventDateTimes: List<EventDateTimeDto>

images: List<ImageDto>

mainImageId: Long

4.DeleteEventRequestDto
Used to trigger event deletion (client → server)

Fields:

eventId: Long

5.DeleteEventResponseDto
Returned after successfully deleting an event

Fields:

success: boolean

message: String

6.EventDateTimeDto
Represents one date and time instance of an event

Fields:

date: LocalDate — must be today or in the future

startTime: LocalTime — optional if allDay = true

endTime: LocalTime — optional if allDay = true

allDay: Boolean — required

7.LocationDto
Represents the physical location of an event (used when type is PLACE)

Fields:

address: String — required

latitude: Double — required

longitude: Double — required

8.ImageDto
Represents metadata for an event image

Fields:

imageId: Long — required (ID of image already uploaded)

isMain: Boolean — optional (true if marked as main image)

📌 Enums Used in DTOs:

Visibility: OPEN, CLOSED

EventType: PLACE, ONLINE

✅ Acceptance Criteria:

All DTOs use standard validation annotations (@NotNull, @Size, @URL, etc.)

DTOs are serializable and compatible with JSON

Proper @Valid usage for nested lists

Ready for mapping with MapStruct or manual conversion

Aligned with Create Event form design

### LINKED ISSUE #145: Create EventDateTimeDtoValidator.
Description:
Create a utility validator class EventDateTimeValidator that enforces business rules for validating and adjusting time-related fields of EventDateTimeDto during event creation and editing.

Context:
The business requirements define strict rules for time selection depending on allDay flag and event date (e.g. no past times, default values, etc). These must be implemented server-side to ensure consistent and secure behavior.

Tasks:

Create class greencity.util.validation.EventDateTimeValidator

Implement method validateAndFill(EventDateTimeDto dto) with the following logic:

a. If dto == null or date == null → throw ValidationException
b. If allDay is true:

Set startTime = 00:00

Set endTime = 12:00
c. If allDay is false:

If date == today and startTime < now → throw ValidationException

If startTime is null → set to now + 1 hour

If endTime is null → set to startTime + 1 hour

If endTime < startTime → throw ValidationException

Annotate the class with @Component for dependency injection.

Inject and use this validator in EventService or EventMapper for each EventDateTimeDto from CreateEventRequestDto and EditEventRequestDto.

Write unit tests for this validator.

Acceptance Criteria:

Validator correctly fills default values for time if missing.

Validator throws exceptions for invalid (past or reversed) times.

Logic respects allDay flag and current date/time.

Unit tests cover all branches and edge cases.

### PR DESCRIPTION
**Pull request by Linney34**

### CODE REVIEW

File: service-api/src/main/java/greencity/dto/event/EventDateTimeDto.java [hunk 1]
```diff
@@ -0,0 +1,17 @@
+package greencity.dto.event;
+
+import jakarta.validation.constraints.NotNull;
+
+import java.time.LocalDate;
+import java.time.LocalTime;
+
+public class EventDateTimeDto {
```
Comment (Kostya-Pav) [inline service-api/src/main/java/greencity/dto/event/EventDateTimeDto.java:17]: 3. Date
Past dates are not allowed
Default: current date
4. Time
Default: current time + 1 hour
User can select:
Start time
End time
or select "All day":
disables time pickers
default times: 00 AM – 12 PM
if today's date: start = current time
Past time not allowed for current date

Comment (Kostya-Pav) [reply service-api/src/main/java/greencity/dto/event/EventDateTimeDto.java:17]: You need to create a validator or check with if blocks

File: service-api/src/main/java/greencity/dto/event/EventImageDto.java [hunk 2]
```diff
@@ -0,0 +1,10 @@
+package greencity.dto.event;
+
+import jakarta.validation.constraints.NotNull;
+
+public class EventImageDto {
+    @NotNull
+    private Long imageId;
+
+    private Boolean isMain;
```
Comment (Kostya-Pav) [inline service-api/src/main/java/greencity/dto/event/EventImageDto.java:19]: Need to check if we have only one main image.

File: service-api/src/main/java/greencity/dto/event/EventDateTimeDto.java (hunk 1 above)
Comment (Linney34) [reply service-api/src/main/java/greencity/dto/event/EventDateTimeDto.java:17]: No need to create validator, I added annotation @FutureOrPresent and it doesnt allow past dates

Comment (Kostya-Pav) [reply service-api/src/main/java/greencity/dto/event/EventDateTimeDto.java:17]: @FutureOrPresent only works for date, but we also need to validate startTime and endTime if allDay is false — especially when the event is today.

**Review by Kostya-Pav — APPROVED** [review]

**Review by MykytaSirobaba — APPROVED** [review]

File: service-api/src/main/java/greencity/dto/event/EventResponseDto.java [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+
+    @NotNull
+    private Long eventId;
+
+    @NotBlank
+    private String title;
+
+    @NotEmpty
+    private LocalDateTime createdAt;
+
```
Comment (MykytaSirobaba) [inline service-api/src/main/java/greencity/dto/event/EventResponseDto.java:45]: More fields

**Review by MaksJv — APPROVED** [review]
